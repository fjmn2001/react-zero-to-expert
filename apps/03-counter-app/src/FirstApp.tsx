import "./index.css"

interface FirstAppProps {
  greeting: string
  subtitle?: string
}

const FirstApp = ({
  greeting,
  subtitle = "I am a subtitle",
}: FirstAppProps) => {
  const numbers = 123
  const user = {
    name: "Francisco",
  }

  return (
    <>
      <h1>{greeting}</h1>
      <p>{subtitle}</p>
      <h1>{numbers}</h1>
      <pre>{JSON.stringify(user)}</pre>
    </>
  )
}

export default FirstApp
