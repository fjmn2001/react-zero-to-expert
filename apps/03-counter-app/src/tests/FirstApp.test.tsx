import { shallow } from "enzyme"

import { describe, expect, test } from "vitest"
import FirstApp from "../FirstApp"

describe("Testing FirstApp", () => {
  test("it should show FirstApp", () => {
    const greeting = "Hi there!"
    const wrapper = shallow(<FirstApp greeting={greeting} />)

    expect(wrapper).toMatchSnapshot()
  })
})
// how remove react from header
