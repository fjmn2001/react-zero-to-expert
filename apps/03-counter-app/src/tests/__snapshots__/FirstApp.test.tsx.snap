// Vitest Snapshot v1

exports[`Testing FirstApp > it should show FirstApp 1`] = `
<Fragment>
  <h1>
    Hi there!
  </h1>
  <p>
    I am a subtitle
  </p>
  <h1>
    123
  </h1>
  <pre>
    {"name":"Francisco"}
  </pre>
</Fragment>
`;
